packages/vm/src/static_analysis.rs

```diff
@@ -1,16 +1,59 @@
 use std::collections::HashSet;
 
+use strum::{AsRefStr, Display, EnumString};
 use wasmer::wasmparser::ExternalKind;
 
 use crate::parsed_wasm::ParsedWasm;
 
-pub const REQUIRED_IBC_EXPORTS: &[&str] = &[
-    "ibc_channel_open",
-    "ibc_channel_connect",
-    "ibc_channel_close",
-    "ibc_packet_receive",
-    "ibc_packet_ack",
-    "ibc_packet_timeout",
+/// An enum containing all available contract entrypoints.
+/// This also provides conversions to and from strings.
+#[derive(PartialEq, Eq, Debug, Clone, Copy, Hash, EnumString, Display, AsRefStr)]
+pub enum Entrypoint {
+    #[strum(serialize = "instantiate")]
+    Instantiate,
+    #[strum(serialize = "execute")]
+    Execute,
+    #[strum(serialize = "migrate")]
+    Migrate,
+    #[strum(serialize = "sudo")]
+    Sudo,
+    #[strum(serialize = "reply")]
+    Reply,
+    #[strum(serialize = "query")]
+    Query,
+    #[strum(serialize = "ibc_channel_open")]
+    IbcChannelOpen,
+    #[strum(serialize = "ibc_channel_connect")]
+    IbcChannelConnect,
+    #[strum(serialize = "ibc_channel_close")]
+    IbcChannelClose,
+    #[strum(serialize = "ibc_packet_receive")]
+    IbcPacketReceive,
+    #[strum(serialize = "ibc_packet_ack")]
+    IbcPacketAck,
+    #[strum(serialize = "ibc_packet_timeout")]
+    IbcPacketTimeout,
+}
+
+// sort entrypoints by their &str representation
+impl PartialOrd for Entrypoint {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+impl Ord for Entrypoint {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.as_ref().cmp(other.as_ref())
+    }
+}
```

Comment:
Oh, love this solution. It allows us to group the enum cases by category in the enum definition (as we do now) instead of having to sort them.

Reply:
Yes, and it also makes sure we can't accidentally misorder them